packages/frontend-plugin-api/src/analytics/Tracker.ts

```diff
@@ -0,0 +1,180 @@
+  constructor(
+    private readonly analyticsApi: AnalyticsApi,
+    private context: AnalyticsContextValue = {
+      routeRef: 'unknown',
+      pluginId: 'root',
+      extension: 'App',
+    },
+  ) {
+    // Only register a single beforeunload event across all trackers.
+    if (!globalEvents.beforeUnloadRegistered) {
+      // Before the page unloads, attempt to capture any deferred navigation
+      // events that haven't yet been captured.
+      addEventListener(
+        'beforeunload',
+        () => {
+          if (globalEvents.mostRecentGatheredNavigation) {
+            this.analyticsApi.captureEvent({
+              ...globalEvents.mostRecentGatheredNavigation,
+              ...globalEvents.mostRecentRoutableExtensionRender,
+            });
+            globalEvents.mostRecentGatheredNavigation = undefined;
+            globalEvents.mostRecentRoutableExtensionRender = undefined;
+          }
+        },
+        { once: true, passive: true },
+      );
+
+      // Prevent duplicate handlers from being registered.
+      globalEvents.beforeUnloadRegistered = true;
+    }
+  }
+
+  setContext(context: AnalyticsContextValue) {
+    this.context = context;
+  }
+
+  captureEvent(
+    action: string,
+    subject: string,
+    {
+      value,
+      attributes,
+    }: { value?: number; attributes?: AnalyticsEventAttributes } = {},
+  ) {
+    // Never pass internal "_routeNodeType" context value.
+    const { _routeNodeType, ...context } = this.context;
+
+    // Never fire the special "_routable-extension-rendered" internal event.
+    if (action === routableExtensionRenderedEvent) {
+      // But keep track of it if we're delaying a `navigate` event for a
+      // a gathered route node type.
+      if (globalEvents.mostRecentGatheredNavigation) {
+        globalEvents.mostRecentRoutableExtensionRender = {
+          context: {
+            ...context,
+            extension: 'App',
+          },
+        };
+      }
+      return;
+    }
+
+    // If we are about to fire a real event, and we have an un-fired gathered
+    // mountpoint navigation on the global store, we need to fire the navigate
+    // event first, so this real event happens accurately after the navigation.
+    if (globalEvents.mostRecentGatheredNavigation) {
+      try {
+        this.analyticsApi.captureEvent({
+          ...globalEvents.mostRecentGatheredNavigation,
+          ...globalEvents.mostRecentRoutableExtensionRender,
+        });
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Error during analytics event capture. %o', e);
+      }
+
+      // Clear the global stores.
+      globalEvents.mostRecentGatheredNavigation = undefined;
+      globalEvents.mostRecentRoutableExtensionRender = undefined;
+    }
+
+    // Never directly fire a navigation event on a gathered route with default
+    // contextual details.
+    if (
+      action === 'navigate' &&
+      _routeNodeType === 'gathered' &&
```

Comment:
Hmm, this is gonna need a bit of a refactor since we'll never get this type of node anymore. Effectively all routes are "gathered" now so I think we might need to apply this logic to all events. I believe that the equivalent in the new system is to check if our current extension doesn't have a `.source` and in that case go down this path.